internal/driver/onvifdiscovery.go

```diff
@@ -189,9 +189,9 @@ func executeRawProbe(conn net.Conn, params netscan.Params) ([]onvif.Device, erro
 
 	var responses []string
 	buf := make([]byte, bufSize)
-	// keep reading from the PacketConn until the read deadline expires or an error occurs
+	//keep reading responses from the connection
```

Comment:
I think you should still keep the `until the read deadline expires or an error occurs` part:

```go
// keep reading responses from the connection until the read deadline expires or an error occurs
```